codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,356 @@
+        )
+        subparser.add_argument(
+            '--gpus', type=int, default=1, help='Default number of GPUs for each worker'
+        )
+        subparser.add_argument(
+            '--memory-mb', type=int, default=2048, help='Default memory (in MB) for each worker'
+        )
+        subparser.add_argument(
+            '--time',
+            type=str,
+            default='10-0',
+            help='Set a limit on the total run time in minutes or days-hours '
+            'of the job allocation. Default to 10-0 (10 days, 0 hours)',
+        )
+        subparser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help='Print out Slurm batch job definition without submitting to Slurm',
+        )
+        subparser.add_argument(
+            '--user', type=str, default=getpass.getuser(), help='User to run the Batch jobs as'
+        )
+        subparser.add_argument(
+            '--password-file',
+            type=str,
+            help='Path to the file containing the username and password '
+            'for logging into the CodaLab worker each on a separate '
+            'line. If not specified, the worker will read from '
+            'environment variable CODALAB_USERNAME and CODALAB_PASSWORD',
+        )
+        subparser.add_argument(
+            '--slurm-work-dir',
+            default='slurm-worker-scratch',
+            help='Directory where to store Slurm batch scripts, logs, etc',
+        )
+
+    def __init__(self, args):
+        super().__init__(args)
+        self.username = self.args.user
+        # A set of newly submitted job id to keep tracking worker status, as worker might not be created right away.
+        self.submitted_jobs = self.load_worker_jobs()
+
+    def load_worker_jobs(self):
+        """
+        Load worker jobs that are created using SlurmWorkerManager and
+        owned by the current user from the Slurm scheduling queue.
+        :return: a set of job id
+        """
+        # Get all the Slurm workers that are submitted by SlurmWorkerManager and owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID:NAME (header won't be included with "--noheader" option)
+        # 1487896,john-job-3157358
+        # 1478830,john-job-5234492
+        submitted_jobs = set()
+        jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '--format', '%A,%j', '--noheader']
+        )
+        for job in jobs.strip().split():
+            job_id, job_name = job.split(',')
+            if job_name.startswith(self.username) and self.args.job_name in job_name:
+                submitted_jobs.add(job_id)
+        return submitted_jobs
+
+    def get_worker_jobs(self):
+        """
+        Return a list of workers in RUNNING and PENDING state.
+        The current Slurm version on NLP cluster is 17.11.13-2, which doesn't have rest api provided.
```

Comment:
Omit mention of NLP cluster? Just say something about how you do this for compatibility with older slurm setups. Should also probably have a note somewhere with the slurm version that this was developed / tested with, just for users' information

Reply:
Make sense.